src/app/modules/verification/verification.service.ts

```diff
@@ -117,25 +122,26 @@ export const getNewOtp = async (
   fieldId: string,
   answer: string,
 ): Promise<void> => {
-  if (isTransactionExpired(transaction.expireAt)) {
+  if (isTransactionExpired(transaction.expireAt!)) {
     throwError(VfnErrors.TransactionNotFound)
   }
   const field = getFieldFromTransaction(transaction, fieldId)
-  if (field === undefined) {
-    throwError('Field not found in transaction', VfnErrors.FieldNotFound)
+  if (!field) {
+    return throwError('Field not found in transaction', VfnErrors.FieldNotFound)
   }
   const { _id: transactionId, formId } = transaction
-  const waitForSeconds = waitToResendOtpSeconds(field.hashCreatedAt)
+  const waitForSeconds = waitToResendOtpSeconds(field.hashCreatedAt!)
   if (waitForSeconds > 0) {
-    throwError(
+    return throwError(
       `Wait for ${waitForSeconds} seconds before requesting for a new otp`,
       VfnErrors.WaitForOtp,
     )
   } else {
     const hashCreatedAt = new Date()
     const otp = generateOtp()
     const hashedOtp = await bcrypt.hash(otp, SALT_ROUNDS)
-    const signedData = formsgSdk.verification.generateSignature({
+
+    const signedData = formsgSdk.verification.generateSignature!({
```

Comment:
is there a bug with the SDK type declarations?

Reply:
Yes, it's fixed in a PR but it's not published yet I think, it's been a while, need to check for breaking changes too haha

Reply:
ok, that sounds like tech debt that we should address when you're back